hw-w2.js

```diff
@@ -0,0 +1,147 @@
+// *** number 8 from Javascript1 week2 ***
+
+// 8.1
+let var1 = 111;
+let var2 = 'aaa';
+let var3 = true;
+let var4 = null;
+
+// 8.2
+console.log('the value of var1 is: ' + var1);
+console.log('the value of var2 is: ' + var2);
+console.log('the value of var3 is: ' + var3);
+console.log('the value of var4 is: ' + var4);
+
+// 8.3
+console.log('the value of var1 is a number);
+console.log('the value of var2 is a string);
+console.log('the value of var3 is a boolean);
+console.log('the value of var4 is an object); 
+
+// 8.4
+console.log('the value of var1 is a ' typeof var1); 
+console.log('the value of var2 is a ' typeof var2); 
+console.log('the value of var3 is a ' typeof var3); 
+console.log('the value of var4 is an ' typeof var4);
+
+// 8.5
+typeof var1 !== typeof var2; //returned: true
+typeof var3 == typeof var4; //returned: false
+
+// 8.6
+// with alert message
+if (typeof var1 !== typeof var4);
+alert('var1 and var4 are not the same type of variables!'); // returned pop-up message: var 1 and var4 are not the same type of variables!
+
+// with console message
+if (typeof var1 !== typeof var4);
+console.log('var1 and var4 are not the same type of variables!'); // returned: var1 and var4 are not the same type of variables!
+
+
+// *** number 9 from Javascript1 week2 ***
+
+// 9: the new value of x would be 1, because the mathematical operator '%' provides with the remainder value when the assigned value is divided by the value at the right side of '%'.
+
+// 9.1 explanatory examples:
+a = 90; //returned 90
+b = a % 9; //returned 0
+c = a % 5; //returned 0
+
+// *** number 10 from Javascript1 week2 ***
+
+// 10: yes, multiple types would be stored in an array.
+
+// 10.1
+myArray = [1, 2, 3, 'hahaha', null, [true, false]]; // returned: Array(6) [ 1, 2, 3, "hahaha", null, (2) […] ]
+
+// 10.2: comparing infinities depends on the type of value. in this case (6/0 === 10/0), it results in 'true' because === operator focuses on the types of the given values, not the values. here both of them is 'NaN'.
```

Comment:
Close, both are in fact `Infinity` which is a pre-defined global variable, you can check this in the console!